coding-challenges.js

```diff
@@ -1,60 +1,118 @@
 // ASSESSMENT 4: JavaScript Coding Practical Questions
 
-// --------------------1) Create a function that takes in an array. Each time the function is run, remove the first item from the array and shuffle the remaining content. If the array is empty, return "The array is empty."
-
-
-
-
-
-var collections = ["purple", "blue", "green", "yellow", "pink"]
+// --------------------
+// // 1) Create a function that takes in an array. 
+// // Each time the function is run, 
+// // remove the first item from the array and 
+// shuffle the remaining content. 
+// If the array is empty, return "The array is empty."
+
+var collections = ["purple", "blue", "green", "yellow", "pink"];
+
+function arrMixer(array) {
+   
+    for (let i = array.length - 1; i > 0; i--) {
+        const newI = Math.floor(Math.random() * (i + 1));
+        [array[i], array[newI]] = [array[newI], array[i]];
+    }
+    if (array.shift() >= 0){
+    }
+ return array;
+ // else {
+    // return 'empty array';
+    // }
```

Comment:
it's not immediately clear what is happening in this code. especially with the line:
`[array[i], array[newI]] = [array[newI], array[i]];`
think about adding comments